src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/DraftReferralTest.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import java.text.SimpleDateFormat
+import java.util.UUID
+
+class DraftReferralTest : IntegrationTestBase() {
+
+  @Test
+  fun `empty input cannot create draft referral`() {
+    assertThrows<RuntimeException> {
+      DraftReferral(Referral())
+    }
+  }
+
+  @Test
+  fun `empty referral id cannot create draft referral`() {
+    val createdDate = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse("2020-12-04 10:42:43")
+
+    assertThrows<RuntimeException> {
+      DraftReferral(Referral(created = createdDate))
+    }
+  }
+
+  @Test
+  fun `empty referral creation date cannot create draft referral`() {
+    val id = UUID.fromString("70d3a47c-d539-4f76-8fc9-1c50e34aea29")
+
+    assertThrows<RuntimeException> {
+      DraftReferral(Referral(id = id))
+    }
+  }
+
+  @Test
+  fun `create draft referral dto from referral`() {
+    val createdDate = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse("2020-12-04 10:42:43")
+    val id = UUID.fromString("70d3a47c-d539-4f76-8fc9-1c50e34aea29")
+    val draftReferral = DraftReferral(Referral(id = id, created = createdDate))
+
+    assertThat(draftReferral.id).isEqualTo("70d3a47c-d539-4f76-8fc9-1c50e34aea29")
+    assertThat(draftReferral.created).isEqualTo("2020-12-04T10:42:43Z")
+  }
+
+  @Test
+  fun `determine json for draft referral dto`() {
```

Comment:
Out of curiosity, are we testing the serialisation to JSON to assert that the correct field formats are used?

Reply:
If yes, what do you think about testing this on unit level by testing the `dto.DraftReferral` construction? That'd avoid having to maintain a JSON payload here for all field changes and avoid a reasonably costly test case.